### PR TITLE
Improve pretty format if a block consists of a single numeric value and exceeds one million.

### LINKED ISSUE #60296: Usability improvement for Pretty format: print human readable number in the case of a single value
```
clickhouse-cloud :) SELECT count() FROM merge(default, '^trace_log_')

SELECT count()
FROM merge(default, '^trace_log_')

Query id: 5bc2e90c-2c69-4152-bfdc-fc3c39b0ab30

┌──────count()─┐
│ 233765663884 │
└──────────────┘
```

If a block consists of a single numeric value, and it exceeds one million, we should print:

```
┌──────count()─┐
│ 233765663884 │ -- 233.77 billion
└──────────────┘
```

This addition should be displayed in gray color, if the output is a terminal.

### PR DESCRIPTION
**Pull request by rogeryk**
### Changelog category:
- Improvement

### Changelog entry:
When output format is Pretty format and a block consists of a single numeric value which exceeds one million,
A readable number will be printed on table right. e.g.
```
┌──────count()─┐
│ 233765663884 │ -- 233.77 billion
└──────────────┘
```

closes #60296

### CODE REVIEW

File: src/Processors/Formats/Impl/PrettyBlockOutputFormat.cpp [hunk 1]
```diff
@@ -410,6 +418,13 @@ void PrettyBlockOutputFormat::writeSuffix()
     }
 }
 
+void PrettyBlockOutputFormat::writeReadableNumberTip(double value)
+{
+    writeCString("\033[90m -- ", out);
```
Comment (alexey-milovidov) [inline src/Processors/Formats/Impl/PrettyBlockOutputFormat.cpp:423]: Only if the output is a terminal.

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit 85a15cd855408f10c862228662b2cb00a3d4b68e with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60379/85a15cd855408f10c862228662b2cb00a3d4b68e/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/60379/85a15cd855408f10c862228662b2cb00a3d4b68e/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Processors/Formats/Impl/PrettyBlockOutputFormat.cpp (hunk 1 above)
Comment (alexey-milovidov) [reply src/Processors/Formats/Impl/PrettyBlockOutputFormat.cpp:423]: See the `bool color` variable, it is always set up to what we need.

**Comment by alexey-milovidov** [comment]
Also, add a setting `output_format_pretty__single_large_number_tip_threshold` with the value 1000000 by default. If the value is 0, the tip is not written. Add it also to `SettingsChangesHistory`.

File: src/Processors/Formats/Impl/PrettyCompactBlockOutputFormat.cpp [hunk 2]
```diff
@@ -171,6 +173,12 @@ void PrettyCompactBlockOutputFormat::writeRow(
     }
 
     writeCString(grid_symbols.bar, out);
+    if (single_numeric_value)
+    {
+        auto value = columns[0]->getFloat64(0);
```
Comment (alexey-milovidov) [inline src/Processors/Formats/Impl/PrettyCompactBlockOutputFormat.cpp:178]: We can put these two lines also inside the `writeReadableNumberTip` function, so there will be less duplication.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**